core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,128 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
             learner_playlist_model.collection_ids if
             learner_playlist_model else [])
 
-    def test_mark_exploration_to_be_played_later(self):
+    def test_adding_subscribed_exploration_to_playlist_does_not_add_again(self):
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(self.user_id), [])
 
-        # Add an exploration to the learner playlist.
+        learner_playlist_services.remove_exploration_from_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Test adding the exploration_id if it is already in
+        # learner_playlist.exploration_ids
+        # Add two explorations to the learner playlist.
         learner_progress_services.add_exp_to_learner_playlist(
             self.user_id, self.EXP_ID_0)
-        self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(self.user_id),
-            [self.EXP_ID_0])
-
-        # Add another exploration.
         learner_progress_services.add_exp_to_learner_playlist(
             self.user_id, self.EXP_ID_1)
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
-        # Add the first exploration in a different position.
+        # Add the first exploration to the second position.
         learner_progress_services.add_exp_to_learner_playlist(
             self.user_id, self.EXP_ID_0,
             position_to_be_inserted=1)
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), [self.EXP_ID_1, self.EXP_ID_0])
 
-        # Add the first exploration in the first position.
-        learner_progress_services.add_exp_to_learner_playlist(
-            self.user_id, self.EXP_ID_0,
-            position_to_be_inserted=0)
-        self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
-
-        # Add an exploration to the in progress list of the user.
```

Comment:
Why is this part removed?

Reply:
I didn't see what it was testing in learner_playlist_services since the file never checks whether an exploration is incomplete.

Reply:
Actually, I would keep it, maybe in its own test. The knowledge that marking the exploration as incomplete doesn't change anything is a good constraint to maintain throughout the future.

Reply:
Got it!